Amazon.IonDotnet/Internals/Text/IonTextWriter.cs

```diff
@@ -180,22 +181,50 @@ public override void WriteInt(BigInteger value)
 
         public override void WriteFloat(double value)
         {
-            this.StartValue();
-            this.textWriter.Write(value);
-            this.CloseValue();
+            if (this.options.JsonDowngrade && (double.IsNaN(value) || double.IsInfinity(value)))
+            {
+                this.WriteNull();
+            }
+            else
+            {
+                this.StartValue();
+                this.textWriter.Write(value);
+                this.CloseValue();
+            }
         }
 
         public override void WriteDecimal(decimal value)
         {
             this.StartValue();
-            this.textWriter.Write(value);
+
+            if (this.options.JsonDowngrade)
+            {
+                var dString = value.ToString(CultureInfo.InvariantCulture);
+                this.textWriter.Write(dString);
+            }
```

Comment:
To confirm, does this just write the decimal as a JSON number?

Looks like `TestDecimal` tests `BigDecimal`; can we make it test `decimal`?

Reply:
readValues will always read the IonDecimal's BigDecimal value, so I will test this in a separate test.